src/components/WatchVideo.vue

```diff
@@ -367,15 +367,29 @@ export default {
             return this.fetchJson(this.apiUrl() + "/streams/" + this.getVideoId());
         },
         async fetchSponsors() {
-            return await this.fetchJson(this.apiUrl() + "/sponsors/" + this.getVideoId(), {
-                category:
-                    '["' +
-                    this.getPreferenceString("selectedSkip", "sponsor,interaction,selfpromo,music_offtopic").replaceAll(
-                        ",",
-                        '","',
-                    ) +
-                    '"]',
+            var selectedSkip = this.getPreferenceString(
+                "selectedSkip",
+                "sponsor,interaction,selfpromo,music_offtopic",
+            ).split(",");
+            const skipOptionsJSON = localStorage.getItem("skipOptions");
```

Comment:
Please use a `getPreference` method here.

Reply:
In this case, I'm going to add a `getPreferencesJSON()` to support loading complex preferences from storage. A matching setter also feels appropriate.